src/main/java/com/dietsheet_server/model/diet/Meal.java

```diff
@@ -42,8 +45,21 @@ public Meal() {
     }
 
     public void updateIngredients(List<Ingredient> newIngredients) {
-        this.ingredients.clear();
-        this.ingredients.addAll(newIngredients);
+        ingredients.forEach(ingredient ->
+                ingredient.getProduct().removeIngredientReference(ingredient));
+        ingredients.clear();
+        newIngredients.forEach(ingredient ->
+                ingredient.getProduct().addIngredientReference(ingredient));
+        ingredients.addAll(newIngredients);
+    }
+
+    @JsonIgnore
+    @Override
+    public List<DietEntity> getParents() {
+        return days
+                .stream()
+                .map(day -> (DietEntity) day)
+                .collect(Collectors.toList());
```

Comment:
O co tu się rozchodzi?

Reply:
Rzutuje Day na DietEntity żeby pasowało do typu zwracanego.

Reply:
Tak się teraz zastanawiam czy ja czegoś tu nie odwalilem xD zerknę na to.

Reply:
Tak się teraz zastanawiam czy ja czegoś tu nie odwalilem xD zerknę na to.